pkg/libinit/root_linux_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package libinit
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/cmdline"
+)
+
+func TestLoadModule(t *testing.T) {
+	var loadedModules []string
+	loader := &InitModuleLoader{
+		Cmdline: cmdline.NewCmdLine(),
+		Prober: func(name, params string) error {
+			loadedModules = append(loadedModules, name)
+			return nil
+		},
+	}
+
+	expectedModules := []string{"test", "something-test"}
+	InstallModules(loader, expectedModules)
+	if !reflect.DeepEqual(expectedModules, loadedModules) {
+		t.Fatalf("failed installing modules. expected %+v got %+v\n", expectedModules, loadedModules)
+	}
+}
+
+func TestModuleConf(t *testing.T) {
+	var toBytes = func(s string) []byte {
+		return bytes.NewBufferString(s).Bytes()
+	}
+	var files = []struct {
+		Name    string
+		Content string
+		Modules []string
+	}{
+		{
+			Name:    "test.conf",
+			Content: `something`,
+			Modules: []string{"something"},
+		},
+		{
+			Name: "test2.conf",
+			Content: `module1
+# not a module
+module2`,
+			Modules: []string{"module1", "module2"},
+		},
+	}
+
+	dir, err := os.MkdirTemp("", "test_moddir")
```

Comment:
I think you can replace this whole block with `T.TempDir` (https://pkg.go.dev/github.com/google/go-cmp/cmp#Diff)